Slurm runner

FlyingFish800 opened the pull request:
Current state of Slurm runner branch. Has added utilities to check state of all docker containers/slurm nodes (--info) and to launch a container on a specific machine. Still need to merge with main and squash commits.

5surim commented:
Also, could you clean up the commit history?
We don't want to have any 'Merge~' commit.
"Merge branch 'main' into Slurm_runner"

5surim commented on README.md:
```diff
@@ -12,6 +12,89 @@ To run docker as a non-root user, run: "sudo chmod 666 /var/run/docker.sock" (re
 Add a new SSH key to your GitHub account: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account?platform=linux
 
+## Running on a Slurm cluster
+This script must be run on a slurm cluster connected to the NFS. It will spawn all simpoints as slurm jobs to run on any node where the docker container is found. If no nodes are found it will attempt launch a container on one of the nodes using this Scarab-Infra repository (which should be on the nfs to launch containers from).
```
Could you move this documentation to 'README.LitzLab.md' since this one is all about the slurm runner on NFS.

5surim commented on README.md:
```diff
@@ -12,6 +12,89 @@ To run docker as a non-root user, run: "sudo chmod 666 /var/run/docker.sock" (re
+## Running on a Slurm cluster
+This script must be run on a slurm cluster connected to the NFS. It will spawn all simpoints as slurm jobs to run on any node where the docker container is found. If no nodes are found it will attempt launch a container on one of the nodes using this Scarab-Infra repository (which should be on the nfs to launch containers from).
+
+### Requirements
```
Could you document the steps to set up Slurm on the nodes for the future use if we re-install OS or so?

5surim commented on README.md:
```diff
@@ -12,6 +12,89 @@ To run docker as a non-root user, run: "sudo chmod 666 /var/run/docker.sock" (re
+### Quickstart
+
+#### 1. Build a container
+Follow the normal instructions for building a container. Make note of A) the path provided via -o, and B) the name placed in apps.list (Ex: docker_traces).
```
Ex: allbench_traces for NFS use just to avoid a confusion.

5surim replied:
Let's support the non-NFS user version and keep docker_traces there.

5surim reviewed (changes requested):
Hi Alex,

Thank you for the PR.
After I run my json, I get many slurm-xxxx.out files inside the path where I cloned this repository. I believe we want to have all the logs inside the experiment path.
Also, all the slurm-xxxx.out files show the following error.

```
cat slurm-6138.out
OCI runtime exec failed: exec failed: unable to start container process: exec: "run_single_simpoint.sh": executable file not found in $PATH: unknown
```

I think `chmod +x run_single_simpoint.sh` you introduced will fix it.

Surim

5surim commented on README.md:
```diff
@@ -12,6 +12,89 @@ To run docker as a non-root user, run: "sudo chmod 666 /var/run/docker.sock" (re
+
+Docker prefix:
+Ex: `"docker_prefix": "app_groupname"`
+This corresponds to the APP_GROUPNAME used in the existing infrastructure, which comes from `apps.list`. This is used when creating docker containers using `./run.sh -o .. -b 2`, which produces containers with `{APP_GROUPNAME}_{USERNAME}`. You can check these container names with `docker ps -a`. This APP_GROUPNAME is provided via docker_prefix so the scarab launch script can detect your container ...
```
allbench instead of docker_traces here again after this documentation is moved to README.LitzLab.md

5surim commented on common/Dockerfile.common:
```diff
@@ -42,6 +42,9 @@ COPY run_scarab_mode_4_allbench.sh /usr/local/bin/run_scarab_mode_4_allbench.sh
 COPY ./common/common_entrypoint.sh /usr/local/bin/common_entrypoint.sh
 
+COPY run_single_simpoint.sh /usr/local/bin/run_single_simpoint.sh
+RUN chmod +x /usr/local/bin/run_single_simpoint.sh
```
Not sure why this didn't work for me.
Just do this outside of the docker container and update scripts worked.

FlyingFish800 replied:
Did you use a pre-existing container? I added these commands to `update_scripts.sh` as well to fix this issue. I'll document this in the ReadMe

5surim replied:
I tested from the very beginning (docker build)

FlyingFish800 replied:
I added a line to the `slurm_runner.py` script to run `chmod +x /usr/local/bin/run_single_simpoint.sh` before a simpoint is run to fix this issue

FlyingFish800 commented:
I moved all the slurm runner documentation to the Litz Lab readme, and wrote a new readme explaining how to install slurm. I also modified the runner script to run `chmod +x run_single_simpoint.sh` before trying to run a simpoint. I also changed the slurm command so that log files are created in a new directory inside the experiment folder named logs.

5surim reviewed (approved):
LGTM

5surim reviewed (approved):
LGTM
